hamilton/customization/default.py

```diff
@@ -0,0 +1,303 @@
+                    f"\n - node_return_type: {PDBDebugger._truncate_repr(node_return_type)}"
+                    f"\n - task_id: {PDBDebugger._truncate_repr(task_id)}"
+                )
+            )
+            pdb.set_trace()
+
+    def run_after_node_execution(
+        self,
+        *,
+        node_name: str,
+        node_tags: Dict[str, Any],
+        node_kwargs: Dict[str, Any],
+        node_return_type: type,
+        result: Any,
+        error: Optional[Exception],
+        success: bool,
+        task_id: Optional[str],
+        **future_kwargs: Any,
+    ):
+        """Executes after a node, whether or not is was successful. Does nothing, just runs pdb.set_trace().
+
+        :param node_name: Name of the node
+        :param node_tags:  Tags of the node
+        :param node_kwargs:  Keyword arguments passed to the node
+        :param node_return_type:  Return type of the node
+        :param result: Result of the node, None if there was an error
+        :param error: Error of the node, None if there was no error
+        :param success:  Whether the node ran successful or not
+        :param task_id: Task ID of the node, if any
+        :param future_kwargs: Additional keyword arguments that may be passed to the hook yet are ignored for now
+        """
+        if should_run_node(node_name, node_tags, self.node_filter):
+            logging.warning(
+                (
+                    f"Placing you in a PDB debugger post execution of node: {node_name}."
+                    "\nYou can access additional node information via the following variables:"
+                    f"\n - node_name: {PDBDebugger._truncate_repr(node_name)}"
+                    f"\n - node_tags: {PDBDebugger._truncate_repr(node_tags)}"
+                    f"\n - node_kwargs: {PDBDebugger._truncate_repr(', '.join(list(node_kwargs.keys())))}"
+                    f"\n - node_return_type: {PDBDebugger._truncate_repr(node_return_type)}"
+                    f"\n - result: {PDBDebugger._truncate_repr(result)}"
+                    f"\n - error: {PDBDebugger._truncate_repr(error)}"
+                    f"\n - success: {PDBDebugger._truncate_repr(success)}"
+                    f"\n - task_id: {PDBDebugger._truncate_repr(task_id)}"
+                )
+            )
+            pdb.set_trace()
+
+    def __init__(
+        self,
+        node_filter: NodeFilter,
+        before: bool = False,
+        during: bool = False,
+        after: bool = False,
+    ):
+        """Creates a PDB debugger. This has three possible modes:
+        1. Before -- places you in a function with (a) node information, and (b) inputs
+        2. During -- runs the node with pdb.run. Note this may not always work or give what you expect as
+            node functions are often wrapped in multiple levels of input modifications/whatnot. That said, it should give you something.
```

Comment:
To get around the partial/wrapped functions we could want to have a "hook" that would wrap functions at graph build time?

Reply:
Yes, maybe, but its also a little complicated. E.G. how does that interact with decorators? 
The PDB debugger added the caveat that its not complete -- the API will be stable, but the implementation might change.

Some other thoughts:
1. Decorators are actually lifecycle methods -- this could fit there
2. Not every function is actually executed (although `@does` might be the only weird case..)
3. With decorators, its usually one node that actually executes the fn, although sometimes they don't

So yeah, probably a cleaner way to do this, but want to push that off till later

Reply:
yes, not for this PR -- but for the future.